### PR TITLE
More mappings for net/minecraft/world/level/levelgen

### PR DESCRIPTION
**Pull request by PiTheGuy**

### CODE REVIEW

File: data/net/minecraft/world/level/levelgen/DensityFunctions.mapping [hunk 1]
```diff
@@ -1,6 +1,11 @@
 CLASS net/minecraft/world/level/levelgen/DensityFunctions
+	METHOD add (Lnet/minecraft/world/level/levelgen/DensityFunction;Lnet/minecraft/world/level/levelgen/DensityFunction;)Lnet/minecraft/world/level/levelgen/DensityFunction;
+		ARG 0 function1
+		ARG 1 function2
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/DensityFunctions.mapping]:
```suggestion
		ARG 0 argument1
		ARG 1 argument2
```
Consistency with `DensityFunctions.Ap2`, and the codec provided by `DensityFunctions.doubleFunctionArgumentCodec`

File: data/net/minecraft/world/level/levelgen/DensityFunctions.mapping [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
 	METHOD blendDensity (Lnet/minecraft/world/level/levelgen/DensityFunction;)Lnet/minecraft/world/level/levelgen/DensityFunction;
 		ARG 0 input
+	METHOD bootstrap (Lnet/minecraft/core/Registry;)Lcom/mojang/serialization/Codec;
+		ARG 0 codec
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/DensityFunctions.mapping]:
```suggestion
		ARG 0 registry
```
This is not a codec, it is the registry of density functions.

File: data/net/minecraft/world/level/levelgen/DensityFunctions.mapping [hunk 3]
```diff
@@ -24,6 +32,15 @@ CLASS net/minecraft/world/level/levelgen/DensityFunctions
 		ARG 0 noiseData
 		ARG 1 xzScale
 		ARG 3 yScale
+	METHOD max (Lnet/minecraft/world/level/levelgen/DensityFunction;Lnet/minecraft/world/level/levelgen/DensityFunction;)Lnet/minecraft/world/level/levelgen/DensityFunction;
+		ARG 0 function1
+		ARG 1 function2
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/DensityFunctions.mapping]:
```suggestion
		ARG 0 argument1
		ARG 1 argument2
```
See note about `Ap2` above.

File: data/net/minecraft/world/level/levelgen/DensityFunctions.mapping [hunk 4]
```diff
@@ -24,6 +32,15 @@ CLASS net/minecraft/world/level/levelgen/DensityFunctions
...
+	METHOD min (Lnet/minecraft/world/level/levelgen/DensityFunction;Lnet/minecraft/world/level/levelgen/DensityFunction;)Lnet/minecraft/world/level/levelgen/DensityFunction;
+		ARG 0 function1
+		ARG 1 function2
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/DensityFunctions.mapping]:
```suggestion
		ARG 0 argument1
		ARG 1 argument2
```
See note about `Ap2` above.

File: data/net/minecraft/world/level/levelgen/DensityFunctions.mapping [hunk 5]
```diff
@@ -24,6 +32,15 @@ CLASS net/minecraft/world/level/levelgen/DensityFunctions
...
+	METHOD mul (Lnet/minecraft/world/level/levelgen/DensityFunction;Lnet/minecraft/world/level/levelgen/DensityFunction;)Lnet/minecraft/world/level/levelgen/DensityFunction;
+		ARG 0 function1
+		ARG 1 function
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/DensityFunctions.mapping]:
```suggestion
		ARG 0 argument1
		ARG 1 argument2
```
See note about `Ap2` above.

File: data/net/minecraft/world/level/levelgen/NoiseRouterData.mapping [hunk 6]
```diff
@@ -1,3 +1,26 @@
...
+	METHOD nether (Lnet/minecraft/world/level/levelgen/NoiseSettings;)Lnet/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises;
+		ARG 0 noiseSettings
+	METHOD noNewCaves (Lnet/minecraft/world/level/levelgen/NoiseSettings;)Lnet/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises;
+		ARG 0 noiseSettings
+	METHOD overworld (Lnet/minecraft/world/level/levelgen/NoiseSettings;Z)Lnet/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises;
+		ARG 0 noiseSettings
+		ARG 1 large
+	METHOD overworldWithNewCaves (Lnet/minecraft/world/level/levelgen/NoiseSettings;Z)Lnet/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises;
+		ARG 0 noiseSettings
+		ARG 1 large
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/NoiseRouterData.mapping]:
```suggestion
		ARG 1 largeBiomes
```

File: data/net/minecraft/world/level/levelgen/NoiseRouterData.mapping [hunk 7]
```diff
@@ -1,3 +1,26 @@
...
+		ARG 0 noiseSettings
+	METHOD end (Lnet/minecraft/world/level/levelgen/NoiseSettings;)Lnet/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises;
+		ARG 0 noiseSettings
+	METHOD nether (Lnet/minecraft/world/level/levelgen/NoiseSettings;)Lnet/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises;
+		ARG 0 noiseSettings
+	METHOD noNewCaves (Lnet/minecraft/world/level/levelgen/NoiseSettings;)Lnet/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises;
+		ARG 0 noiseSettings
+	METHOD overworld (Lnet/minecraft/world/level/levelgen/NoiseSettings;Z)Lnet/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises;
+		ARG 0 noiseSettings
+		ARG 1 large
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/NoiseRouterData.mapping]:
```suggestion
		ARG 1 largeBiomes
```
It's used to select between density functions, i.e. `FACTOR` vs. `FACTOR_LARGE`, but the input to this is the "large biomes" option.

File: data/net/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises.mapping [hunk 8]
```diff
@@ -0,0 +1,3 @@
+CLASS net/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises
+	METHOD mapAll (Lnet/minecraft/world/level/levelgen/DensityFunction$Visitor;)Lnet/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises;
+		ARG 1 visitors
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/NoiseRouterWithOnlyNoises.mapping]:
```suggestion
		ARG 1 visitor
```

File: data/net/minecraft/world/level/levelgen/SurfaceSystem.mapping [hunk 9]
```diff
@@ -9,3 +9,8 @@ CLASS net/minecraft/world/level/levelgen/SurfaceSystem
 		ARG 1 location
 	METHOD isStone (Lnet/minecraft/world/level/block/state/BlockState;)Z
 		ARG 1 state
+	METHOD topMaterial (Lnet/minecraft/world/level/levelgen/SurfaceRules$RuleSource;Lnet/minecraft/world/level/levelgen/carver/CarvingContext;Ljava/util/function/Function;Lnet/minecraft/world/level/chunk/ChunkAccess;Lnet/minecraft/world/level/levelgen/NoiseChunk;Lnet/minecraft/core/BlockPos;Z)Ljava/util/Optional;
+		ARG 2 context
+		ARG 3 biomeGetter
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/SurfaceSystem.mapping:14]:
```suggestion
		ARG 3 biomeAccessor
```
This ultimately routes through [`WorldCarver#carveBlock`](https://github.com/ParchmentMC/Parchment/blob/5a72c7dcea6ee1a6f810f11be0d676351a2d9691/data/net/minecraft/world/level/levelgen/carver/WorldCarver.mapping#L33) which already names it thusly.

File: data/net/minecraft/world/level/levelgen/blending/BlendingData.mapping [hunk 10]
```diff
@@ -2,5 +2,12 @@ CLASS net/minecraft/world/level/levelgen/blending/BlendingData
 	METHOD <init> (ZLjava/util/Optional;)V
 		ARG 1 oldNoise
 		ARG 2 heights
+	METHOD getDensityColumn (Lnet/minecraft/world/level/chunk/ChunkAccess;III)[D
+		ARG 0 chunkAccess
+	METHOD isGround (Lnet/minecraft/world/level/chunk/ChunkAccess;Lnet/minecraft/core/BlockPos;)Z
+		ARG 0 chunkAccess
```
Comment (alcatrazEscapee) [inline data/net/minecraft/world/level/levelgen/blending/BlendingData.mapping]:
```suggestion
		ARG 0 chunk
	METHOD isGround (Lnet/minecraft/world/level/chunk/ChunkAccess;Lnet/minecraft/core/BlockPos;)Z
		ARG 0 chunk
```
There's not really a reason these need to be typed as per the interface, and there's precedent for using `chunk`, for instance in [`ChunkGenerator`](https://github.com/ParchmentMC/Parchment/blob/versions/1.18.x/data/net/minecraft/world/level/chunk/ChunkGenerator.mapping) methods.

**Review by alcatrazEscapee — CHANGES_REQUESTED** [review]

File: data/net/minecraft/world/level/levelgen/SurfaceSystem.mapping (hunk 9 above)
Comment (PiTheGuy) [reply data/net/minecraft/world/level/levelgen/SurfaceSystem.mapping:14]: Yeah, but you're also creating a new `SurfaceRules.Context` object in which that parameter is assigned to a field named `biomeGetter`. This way copies existing Mojmaps.

Comment (alcatrazEscapee) [reply data/net/minecraft/world/level/levelgen/SurfaceSystem.mapping:14]: Oh, I didn't realize that. That mapping pre-dates surface rules then, could you change `WorldCarver#carveBlock` to also be `biomeGetter` then?

Comment (PiTheGuy) [reply data/net/minecraft/world/level/levelgen/SurfaceSystem.mapping:14]: Done

**Review by alcatrazEscapee — APPROVED** [review]

**Pull request merged**